### ISSUE TITLE
Dutch language

### ISSUE THREAD

**Issue by Bluesbroeder** [th1efx/AutoKit-CarPlay#1]
Hi as discussed on Carplay adapter converter, can you replace the English language into Dutch, thanks a lot.
Hope in the attached document all is translated, otherwise let me know.

[vertaling NL.docx](https://github.com/th1efx/AutoKit-CarPlay/files/6228827/vertaling.NL.docx)

**Comment by Bluesbroeder** [comment]
Hi, worked by myself to translate into Dutch. Almost finished bau have some questions:
1) At settings: "language" I cannot find for translation and is still "language". Do you know where I can change this in the Dutch language?
2) Left upperside you have "Back" is this a button or text what I can change?
3) Can I change also the location of the Frame (with Iphone name) and "search frame". This is not in the middle of my Volvo Sensu screen. Probably it is not 800 px. 
Thanks!

**Comment by th1efx** [comment]
Hi, 

1. you can’t change this, I had similar issue I believe it is packed somewhere in the firmware, I looked -could not find it. If you find it let me know. I had originally wanted to remove CMID and Version from Settings Menu but could not due to limitation above. If you find a work around let me know. 

2) this is a image, you will need to
To use ludwigs image extractor php script, extract the .dll file use an editor like paint.net to change the text in both images. Then repack using Ludwigs dll packer. You are limited to the dimensions of the existing file. If it’s 120x120 then it needs to be the same 121x120 will error out. Keep that in mind, re-save as 32bit png

3) yes, in ui.script (this is the 800x480 layout) open with a text editor you will need to update the X corordinates for the list box in 3 sections, paired devices home (with paired), home no paired devices and bt search results, I believe these are the 3 list boxes, you may also need to adjust the x cordinates for the up and down buttons in the search results. 

Once you have made the changes you the  tarball the whole thing.

**Comment by Bluesbroeder** [comment]
Point 3: done and looks now perfect. 
Now up to point 2: For that I need some support, I'am a newbie :)
How do I use the Image extrator php script and the DLL packer and where can I find it?
I found : php UI_DLL.php, is this the extractor?
And: php UI_DLL.php pack, which I could not find at Ludwigs site.
Thanks!

**Comment by th1efx** [comment]
#3 - you may also need to move the blue bar left by how ever many pixels you moved the list box. You will see this when you highlight a device

#2 download PHP for Windows and install it, copy over the 2 .dll files into the php directory where you installed php, open command prompt and navigate to the php installation directory where php.exe is located, run command: php.exe UI_DLL.php this will unpack all the files into a directory in the php folder. Go in there and make the changes to the images. Then run the same command again but with pack: php.exe UI_DLL.php pack and this will repack the images into both dll files in a folder called “new” then drop over and tarball them again

**Comment by Bluesbroeder** [comment]
Sorry to bother you again. 
PHP is installed but I have problems with the configuration file (.ini). After the command as described above I received the following message:
**Please ensure that gd extension is installed and added inside php.ini

Windows php.ini example:
extension_dir="ext"
extension=php_gd.dll**

Checked it: both are released (removal of ; before the code)  and php_gd.dll is present in "ext".
Attached is my ini file (zipped) hopefully you can advise.

Thanks a lot.

[php.zip](https://github.com/th1efx/AutoKit-CarPlay/files/6289564/php.zip)

**Comment by th1efx** [comment]
I have been on MacRumors contributing to the conversation and offering support. Suddenly I was suspended. 

As you are aware I spent countless hours over a week putting together a ecstatically pleasing theme to make these ugly devices better and spent time engaging in productive conversations in the only thread about hacking this device online.
I have been working tirelessly the past 2 weeks to reverse engineer with the help of Ludwigs custom firmware 3 core files which are the main workhorse of this dongle. I even went as far to purchase a 2nd adapter for development purpose. 
I was able to pull apart some binary files decompile and recompile the code that could allow changes in the firmware. I was able to confirm this by actually editing the “Settings Menu” and even adding toggles for the settings you see at 192.168.50.2 contained in the firmware file onto the screen! Something I was not able to do in the ui theme file as I mentioned this was hard coded into the firmware.
I came here on Friday afternoon to post my findings and post an update to the UI theme with the new Settings screen. We have a lot more access to rebuild and access the hardware then I originally thought. 
As I was in the process of writing my post I received a message from the Moderator who suspended me and then immediately blocked me from even replying to him.

I was informed: 

My theme is not welcome here and MacRumors does not approve of my “self promotion” - I am not sure what this means. I don’t receive monetary compensation for anything I do here and all I do is come here to help and answer questions in this thread to make this device less suck and contribute to this project. 
I have decided public development of the UI THEME will halt and new firmware findings will be kept private and used to further develop and attempt to fix my own bugs for my vehicles and not released to public.
I wish everyone the best, the Original UI theme is still available for download on GitHub but I will no longer support it publicly with bugs or improvements.
It’s power trippers like these mods who cause developers like us to leave and abandon these projects. Further digging into this joke forum I see many people that contribute here are banned or suspended by mods. 
I have to much to offer and I will not be feeding this community forum with any information further, it’s unfortunate that there is no other active community working on this dongle and we have mods runs this board like a dictatorship with absurd rules — but it is what it is.
